pkg/build/api/types.go

```diff
@@ -228,8 +228,58 @@ type BuildStatus struct {
 
 	// Output describes the Docker image the build has produced.
 	Output BuildStatusOutput
+
+	// Steps contains details about each build step
+	Steps Steps
+}
+
+// Steps is a slice of build steps that have been recorded.
+type Steps []StepInfo
+
+// RecordInfo records details about each build step
+func (s *Steps) RecordInfo(stepName StepName, startTime unversioned.Time, stopTime unversioned.Time) {
```

Comment:
@csrwng 
@bparees had asked me to add it as a method onto the type in the source-to-image pull request, so that seemed like what should happen here also.  Was just trying to be consistent between the code bases.  Maybe Ben can weigh in here also.